### PR TITLE
61 admin items

### PR DESCRIPTION
**Pull request by mdorrance**
As an Admin, when you visit "/admin/products" you can see all products with an edit button that links to the edit product form. The name of the product also links to the product show page.

Add breadcrumbs to top of admin/products, admin/products/new and admin/products/:id/edit

### CODE REVIEW

File: spec/features/admin_can_view_all_products_spec.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      description = find("#product_form_description").value
+      expect(description).to eq("This is the description for plant 1")
+      expect(page).to have_content("Price")
+      price = find("#product_form_price").value
+      expect(price).to eq("19.99")
+      expect(page).to have_content("Status")
+      status = find("#product_form_status").value
+      category = find("#product_form_category").value
+      expect(page).to have_content("Plants")
+      expect(page).to have_content("Active")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      expect(page).to have_content("Image Url")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      url = find("#product_form_image_url").value
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      expect(url).to eq("plants/plant-2.jpg")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      expect(page).to have_button("Edit Product")
+
+      find('input[type="text"][name*="name"]').set("Richard")
+      find('input[type="text"][name*="description"]').set("A boat.")
+      find('input[type="text"][name*="price"]').set("12.99")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      status = find("#product_form_status").value
+      category = find("#product_form_category").value
+      expect(page).to have_content("Plants")
+      expect(page).to have_content("Active")
+      expect(page).to have_content("Image Url")
+      url = find("#product_form_image_url").value
+      expect(url).to eq("plants/plant-2.jpg")
+      expect(page).to have_button("Edit Product")
+
+      find('input[type="text"][name*="name"]').set("Richard")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      find('input[type="text"][name*="description"]').set("A boat.")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+      expect(price).to eq("19.99")
+      expect(page).to have_content("Status")
+      status = find("#product_form_status").value
+      category = find("#product_form_category").value
+      expect(page).to have_content("Plants")
+      expect(page).to have_content("Active")
+      expect(page).to have_content("Image Url")
+      url = find("#product_form_image_url").value
+      expect(url).to eq("plants/plant-2.jpg")
+      expect(page).to have_button("Edit Product")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+      find('input[type="text"][name*="name"]').set("Richard")
+      find('input[type="text"][name*="description"]').set("A boat.")
+      find('input[type="text"][name*="price"]').set("12.99")
+      find('input[type="text"][name*="image_url"]').set("")
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+      select "Food", from: "product[category_id]"
+      select "Retired", from: "product[status]"
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+      expect(page).to have_content("Image Url")
+      url = find("#product_form_image_url").value
+      expect(url).to eq("plants/plant-2.jpg")
+      expect(page).to have_button("Edit Product")
+
+      find('input[type="text"][name*="name"]').set("Richard")
+      find('input[type="text"][name*="description"]').set("A boat.")
+      find('input[type="text"][name*="price"]').set("12.99")
+      find('input[type="text"][name*="image_url"]').set("")
+      select "Food", from: "product[category_id]"
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+      select "Retired", from: "product[status]"
+      click_button "Edit Product"
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+      expect(current_path).to eq(admin_products_path)
+      within("tr", text: "Richard") do
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: spec/features/admin_can_view_all_products_spec.rb [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+      find('input[type="text"][name*="name"]').set("Richard")
+      find('input[type="text"][name*="description"]').set("A boat.")
+      find('input[type="text"][name*="price"]').set("12.99")
+      find('input[type="text"][name*="image_url"]').set("")
+      select "Food", from: "product[category_id]"
+      select "Retired", from: "product[status]"
+      click_button "Edit Product"
+
+      expect(current_path).to eq(admin_products_path)
```
Comment (mdorrance) [inline spec/features/admin_can_view_all_products_spec.rb]: Corrected.

File: app/assets/stylesheets/custom.css.scss [hunk 15]
```diff
@@ -77,7 +77,15 @@ h6 {
 .img-hover:hover {
   opacity: 0.8;
 }
-
+.img-small {
+  width: 30%;
+}
```
Comment (mdorrance) [inline app/assets/stylesheets/custom.css.scss]: Corrected.

File: app/assets/stylesheets/custom.css.scss [hunk 16]
```diff
@@ -77,7 +77,15 @@ h6 {
...
+.word-length {
+  white-space: nowrap;
```
Comment (mdorrance) [inline app/assets/stylesheets/custom.css.scss]: Corrected.

**Comment by sabondano** [comment]
Looks good @mdorrance.

**Pull request merged**